rpc/api/miner.go

```diff
@@ -126,7 +126,7 @@ func (self *minerApi) SetExtra(req *shared.Request) (interface{}, error) {
 		return nil, err
 	}
 
-	if uint64(len(args.Data)) > params.MaximumExtraDataSize.Uint64()*2 {
+	if uint64(len(args.Data)) > params.MaximumExtraDataSize.Uint64() {
```

Comment:
Ok, hopefully this is what you mean. :) It is now set up so that miner.setExtra will accept either hex  or a string as the input (32 bytes worth of either).